### PR TITLE
[mieb] Task filtering by modality supported by models

### LINKED ISSUE #1632: [mieb] Error message for unsupported models is unclear
When trying to run e.g. moco on a task it does not support one gets the below error. I think we should catch this based on the metadata of the task (and maybe a new metadata field in models in terms of what they support / reuse the existing fields) to catch such errors and have a clean error message that it is not supported by this model

```
ERROR:mteb.evaluation.MTEB:Error while evaluating VidoreSyntheticDocQAAIRetrieval: mocov3_loader.<locals>.MOCOv3Wrapper.get_text_embeddings() got an unexpected keyword argument 'task_name'
Traceback (most recent call last):
  File "/data/niklas/mieb/mteb/scripts/run_mieb.py", line 100, in <module>
    results = evaluation.run(model, output_folder="/data/niklas/mieb/results-mieb-final", batch_size=8, overwrite_results=True)
  File "/data/niklas/mieb/mteb/mteb/evaluation/MTEB.py", line 464, in run
    raise e
  File "/data/niklas/mieb/mteb/mteb/evaluation/MTEB.py", line 425, in run
    results, tick, tock = self._run_eval(
  File "/data/niklas/mieb/mteb/mteb/evaluation/MTEB.py", line 300, in _run_eval
    results = task.evaluate(
  File "/data/niklas/mieb/mteb/mteb/abstasks/Image/AbsTaskAny2AnyRetrieval.py", line 269, in evaluate
    scores[hf_subset] = self._evaluate_subset(
  File "/data/niklas/mieb/mteb/mteb/abstasks/Image/AbsTaskAny2AnyRetrieval.py", line 278, in _evaluate_subset
    results = retriever(corpus, queries)
  File "/data/niklas/mieb/mteb/mteb/evaluation/evaluators/Image/Any2AnyRetrievalEvaluator.py", line 308, in __call__
    return self.retriever.search(
  File "/data/niklas/mieb/mteb/mteb/evaluation/evaluators/Image/Any2AnyRetrievalEvaluator.py", line 127, in search
    query_embeddings = self.model.get_text_embeddings(
TypeError: mocov3_loader.<locals>.MOCOv3Wrapper.get_text_embeddings() got an unexpected keyword argument 'task_name'
```

A current workaround is sth like the below to skip them
```
if ("moco" in model_name) or ("dinov2" in model_name):
    tasks = [task for task in tasks if "t" not in task.metadata.category]
```

### PR DESCRIPTION
**Pull request by isaac-chung**
Fixes #1632 

- Added ModelMeta.modalities, which defaults to `["text"]` to be backwards compatible. VLMs will have `["image","text"]`.
- Added logic in `MTEB.run` for exact match between task and model modalities
- Added modality model meta to relevant models

Here is the new log, showing that moco only supports images.
```
## Evaluating 1 tasks:
Any2AnyRetrieval
    - VidoreSyntheticDocQAAIRetrieval, t2i


INFO:mteb.evaluation.MTEB:

********************** Evaluating VidoreSyntheticDocQAAIRetrieval **********************
INFO:mteb.evaluation.MTEB:nyu-visionx/moco-v3-vit-b only supports ['image'], but the task modalities are ['text', 'image'].
```

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

cc @Muennighoff @gowitheflow-1998

### CODE REVIEW

File: mteb/model_meta.py [hunk 1]
```diff
@@ -94,6 +96,7 @@ class ModelMeta(BaseModel):
     similarity_fn_name: DISTANCE_METRICS | None = None
     use_instuctions: bool | None = None
     zero_shot_benchmarks: list[str] | None = None
+    modalities: str = "t"
```
Comment (KennethEnevoldsen) [inline mteb/model_meta.py:99]:
```suggestion
    modalities: list[Literal["image", "text"]] = ["image", "text"]
```

Comment (KennethEnevoldsen) [reply mteb/model_meta.py:99]: To make it align with the TaskMetadata

**Comment by KennethEnevoldsen** [comment]
I Would imagine the ideal message to be:

```
INFO:mteb.evaluation.MTEB:nyu-visionx/moco-v3-vit-b only supports the modalities ["images"], but the task category is t2i and include both the modalities ["text", "image"].
```

**Comment by isaac-chung** [comment]
@KennethEnevoldsen thanks but the PR was still in draft, and we had the same thought! It should be ready now.

**Review by gowitheflow-1998 — APPROVED** [review]

**Comment by isaac-chung** [comment]
Thanks for the quick reviews folks. Merry Christmas! 🎄🎁

**Pull request merged**